EDU-17911: Update "FastStore versions and support levels" (#2568)
vtexdocs/dev-portal-content: docs/faststore/docs/getting-started/faststore-versions-and-support-levels.mdx

---
title: "FastStore versions and support levels"
excerpt: "Understand FastStore versions, support levels, and guidance for upgrading."

> updatedAt: "2026-03-26T17:33:35.112Z"

Author: ```suggestion
updatedAt: "2026-04-20T17:33:35.112Z"
```

## FastStore v2 and v3

FastStore v2 and v3 share the same core architecture and introduced major improvements over FastStore v1, such as:

- **Simplified updates:** Stores are monitored for version updates with minimal friction, including automatic non-critical updates and a more efficient integration process.
- **Separation of VTEX-managed code and store customizations:** Clear segmentation between VTEX and custom code to help in the debug process.
- **CMS-first approach:** Headless CMS manages content updates (texts, images, links) while behavioral and UI component changes remain code-based, ensuring a structured workflow.
- **Optimized performance and quality monitoring:** Ongoing performance tracking with proactive safeguards to prevent deployments that don't meet Lighthouse and quality standards.
- **Semantic versioning:** FastStore v2 and v3 use [Semantic Versioning](https://semver.org/) for managing releases:
  - **Majors:** Introduce breaking changes, including incompatible API updates.
  - **Minors:** Introduce backward-compatible features.
  - **Patches:** Include backward-compatible bug fixes.

## FastStore v4

> FastStore v4 is the latest version, building on the same core architecture as v2 and v3 while introducing important updates focused on performance, dependency management, and localization. The key changes in v4 are:

Author: ```suggestion
[FastStore v4](https://developers.vtex.com/docs/guides/faststore/getting-started-upgrading-faststore-to-v4) is the latest version, building on the same core architecture as v2 and v3 while introducing important updates focused on performance, dependency management, and localization. The key changes in v4 are:
```

> - **Centralized dependency management:** Next.js and other core dependencies are now managed by `@faststore/cli`, ensuring consistency across projects, reducing dependency conflicts, and simplifying long-term maintenance. Next.js is updated to version 16.

Reviewer: ```suggestion
- **Centralized dependency management:** Next.js and other core dependencies are now managed by [`@faststore/cli`](https://developers.vtex.com/docs/guides/faststore/developer-tools-faststore-cli), ensuring consistency across projects, reducing dependency conflicts, and simplifying long-term maintenance. Next.js is updated to version 16.
```

| **New projects** | Not supported.                                                                                | Not available for new projects. Existing v2 stores should upgrade to v4.       | Not available for new projects. Existing v3 stores should upgrade to v4.      | Recommended for all new storefront implementations. |

If you have any questions about the level of support provided for different FastStore versions, [open a ticket](https://help.vtex.com/en/support) with the VTEX Support team.

## Upgrade between FastStore versions

>⚠️ Before upgrading to a new FastStore version, contact VTEX to ensure compatibility and a smooth transition. For assistance, [open a ticket](https://help.vtex.com/en/support) with the VTEX Support team.

For FastStore projects hosted in the `vtex-sites` organization and not using WebOps ([FastStore v1](https://v1.faststore.dev/docs)), open a ticket with [VTEX Support](https://supporticket.vtex.com/support) to move your project to your account's GitHub organization and update it to v4.

> For stores using FastStore v2 hosted in the account's GitHub organization that need to upgrade to v4, follow the guide [Upgrading FastStore to v4](/tbd).

Author: ```suggestion
For stores using FastStore v2 hosted in the account's GitHub organization that need to upgrade to v4, follow the guide [Upgrading FastStore to v4](https://developers.vtex.com/docs/guides/faststore/getting-started-upgrading-faststore-to-v4).
```

> For stores using FastStore v3 that need to upgrade to v4, follow the guide [Upgrading FastStore to v4](/tbd).

Author: ```suggestion
For stores using FastStore v3 that need to upgrade to v4, follow the guide [Upgrading FastStore to v4](https://developers.vtex.com/docs/guides/faststore/getting-started-upgrading-faststore-to-v4).
```
